common/src/main/java/com/microsoft/identity/common/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -39,17 +86,735 @@ class NativeAuthMsalController : BaseNativeAuthController() {
     companion object {
         private val TAG = NativeAuthMsalController::class.java.simpleName
     }
-    
+
+    /**
+     * Acts as the starting point for the Sign In flow.
+     * Starts off with the sign in initiate call, and returns any error results if produced.
+     * If successful, continues ot the sign in challenge call, again returning any error results if produced.
+     * If the challenge endpoint returns a password required result,
+     * a call to the token endpoint will be made with the password provided in the SignInStartCommandParameters,
+     * again returning any error results, or returning the token if the call succeeds.
+     * If a token is acquired, it is also cached locally as part of this flow.
+     */
+    fun signInStart(parameters: SignInStartCommandParameters): SignInStartCommandResult {
+        LogSession.logMethodCall(TAG, "${TAG}.signInStart")
+
+        try {
+            val oAuth2Strategy = createOAuth2Strategy(parameters)
+
+            val usePassword = parameters is SignInStartUsingPasswordCommandParameters
+
+            var parametersWithScopes: SignInStartUsingPasswordCommandParameters? = null
+            if (usePassword) {
+                val mergedScopes = addDefaultScopes(parameters.scopes)
+                parametersWithScopes = CommandUtil.createSignInStartCommandParametersWithScopes(
+                    parameters as SignInStartUsingPasswordCommandParameters,
+                    mergedScopes
+                )
+            }
+
+            val initiateApiResult = performSignInInitiateCall(
+                oAuth2Strategy = oAuth2Strategy,
+                parameters = parameters // Downcast, because we don't need the password in this API call
+            )
+
+            return processSignInInitiateApiResult(
+                initiateApiResult = initiateApiResult,
+                oAuth2Strategy = oAuth2Strategy,
+                parametersWithScopes = parametersWithScopes, // Include the scopes, to send to /token endpoint in case of password flow
```

Comment:
What do we mean by in case of? We should be certain in our comments about which flow we are exercising.

An overall comment - the way the code has been structured in this method is confusing. I'd recommend having two separate methods for this i.e. `signInStartWithPassword` and `signInStartWithCode`

Reply:
We need more comments, I agree.
I don't agree re. splitting it - the two methods are 90% the same, so not DRY to separate them out. 
We can consider future refactoring, but given the deadline let's not make that in scope of this PR.